MiniApp/Classes/core/Storage/MiniAppKeyChain.swift

```diff
@@ -74,11 +76,78 @@ import Foundation
         return supportedPermissionList
     }
 
-    private func write(keys: KeysDictionary) {
+    private func writePermissionInfo(keys: KeysDictionary) {
         guard let data = try? JSONEncoder().encode(keys) else {
             return
         }
+        setValueInKeyChain(data: data)
+    }
+
+    private func retrieveAllPermissions() -> KeysDictionary? {
+        guard let storedData = getAllKeysFromKeychain() else {
+            return nil
+        }
+
+        guard let keys = ResponseDecoder.decode(decodeType: KeysDictionary.self, data: storedData) else {
+            return nil
+        }
+
+        return keys
+    }
+
+    // MARK: - Cache Verifier Methods
+    internal func setCacheInfo(key: String, for keyId: String) {
+        var keysDic = getAllCacheKeys()
+        guard keysDic?[keyId] == nil else {
+            return // key exists
+        }
+
+        if keysDic != nil {
+            keysDic?[keyId] = key
+        } else {
+            keysDic = [keyId: key]
+        }
+
+        if let keys = keysDic {
+            writeCacheInfo(keys: keys)
+        }
+    }
+
+    internal func getCacheInfo(for keyId: String) -> String? {
+        return getAllCacheKeys()?[keyId]
+    }
+
+    private func getAllCacheKeys() -> CacheVerifierKeysDictionary? {
+        guard let storedData = getAllKeysFromKeychain() else {
+            return nil
+        }
 
+        guard let keys = try? JSONSerialization.jsonObject(with: storedData, options: []) as? CacheVerifierKeysDictionary else {
+            return nil
+        }
+
+        return keys
+    }
+
+    private func writeCacheInfo(keys: CacheVerifierKeysDictionary) {
+        guard let data = try? JSONEncoder().encode(keys) else {
```

Comment:
Seems like this method is a duplicate of `writePermissionInfo`. Could you use a generic write method instead? For example:
```
private func write<T:Encodable>(keys: [String: T])
```

Reply:
Done